src/libproxychains.c

```diff
@@ -74,15 +74,19 @@ static int init_l = 0;
 
 static inline void get_chain_data(proxy_data * pd, unsigned int *proxy_count, chain_type * ct);
 
-static void* load_sym(char* symname, void* proxyfunc) {
+typedef void (*func_t)(void);
```

Comment:
what you're doing here is not just ugly, it's outright wrong. the functions we're dealing with have all different prototypes so we would need the right function pointer type for every single function.
thankfully according to the standard, all pointers can be cast to void* legally (and only to void*).